test/Microsoft.AspNetCore.Sockets.Tests/TestWebSocketConnectionFeature.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Buffers;
+using System.Buffers.Pools;
+using System.IO.Pipelines;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.WebSockets.Internal;
+using Microsoft.Extensions.WebSockets.Internal;
+
+namespace Microsoft.AspNetCore.Sockets.Tests
+{
+    internal class TestWebSocketConnectionFeature : IHttpWebSocketConnectionFeature
+    {
+        public bool IsWebSocketRequest => true;
+
+        public WebSocketConnection Client { get; private set; }
+
+        public ValueTask<IWebSocketConnection> AcceptWebSocketConnectionAsync(WebSocketAcceptContext context)
+        {
+            var pipelineFactory = new PipeFactory(ManagedBufferPool.Shared);
```

Comment:
Well, we can't quite do that, but we do need to handle disposing the factory. The `IWebSocketConnection`s need the `PipeFactory` to stay alive while they run. I'll take a look at where we could dispose them. To be honest though, if we never disposed them it'd be fine. The only resource they hold is the buffer pool and the one we're using in this case is a global one which lives for the life of the app and is never Disposed (which I used because this is test code).